### PR TITLE
Install extensions from config #55

### LINKED ISSUE #55: Autoinstall extensions from Git
We should allow specifying an `extensions` option in config.yaml that will autoinstall extensions correctly.

```yaml
# Autoinstall the following extensions
# (Note that all directories in extensions will be loaded; this is only for initial provisioning)
extensions:
  memcache: Chassis/memcache # mapped to https://github.com/Chassis/memcache.git
  xhprof: https://github.com/Chassis/xhprof.git
```

### PR DESCRIPTION
**Pull request by shadyvb**
Resolves #55

### CODE REVIEW

File: puppet/chassis.rb [hunk 1]
```diff
@@ -85,4 +85,20 @@ def self.normalize_config(config)
 	def self.config
 		self.normalize_config(self.load_config())
 	end
+
+	def self.install_extensions(config)
+	    # Install extensions listed in config
+        if config["extensions"]
```
Comment (rmccue) [inline puppet/chassis.rb]: Looks like this needs spaces -> tabs

File: puppet/chassis.rb [hunk 2]
```diff
@@ -85,4 +85,20 @@ def self.normalize_config(config)
...
+	def self.install_extensions(config)
+		# Install extensions listed in config
+		if config["extensions"]
+			config["extensions"].each do |ext|
+				begin
+					repo = /^(git@|https:)/.match(ext) ? ext : 'https://github.com/' + ext
+					folder = @@dir + '/extensions/' + ext.split('/').last.gsub(/.git$/, '')
+
+					if ! File.exist?( folder )
+						system("git clone %s %s --recursive" % [repo, folder] )
```
Comment (rmccue) [inline puppet/chassis.rb]: Does this output to the terminal?

Comment (shadyvb) [inline puppet/chassis.rb]: Yup

Comment (rmccue) [inline puppet/chassis.rb]: neato!

**Comment by rmccue** [comment]
✨ 💯 🌴

**Comment by BronsonQuick** [comment]
💥

**Pull request merged**